Sources/StreamChat/Utils/StateBuilder.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A builder for objects requiring @MainActor.
+final class StateBuilder<State> {
+    private var builder: (@MainActor() -> State)?
+    
+    init(builder: (@escaping @MainActor() -> State)) {
+        self.builder = builder
+    }
+    
+    @MainActor func build() -> State {
+        guard let builder else {
+            fatalError("Calling build multiple times for \(State.self)")
```

Comment:
No fatal errors please 😅  We avoid crashing the SDK by ourselves.

Reply:
I will need to think about this one a bit. I only added the part of setting it to nil since then the closure can release captured values. Hmm.

Reply:
I found an alternative solution as well. Would avoid having this type at all.

Reply:
Alternative did not work, I deleted the optional part from there since most of the time we capture things which are alive all the time anyway.